### DISCUSSION TITLE
Using different plugins for the same file pattern

### DISCUSSION THREAD

**Discussion by yeonjuan** [eslint/eslint#18808] [Q&A]
Hi, I was checking on [an issue](https://github.com/yeonjuan/html-eslint/issues/211) in the html-eslint plugin and have a question.
When using different plugins for HTML files, as shown below, currently only the setting at the back is always applied and the setting at the front is ignored. In the following cases, only the eslintTailwindCSS plugin is applied and html-eslint is ignored.
Is this intended behavior and not a bug?

```ts
import eslintHTML from "@html-eslint/eslint-plugin";
import eslintTailwindCSS from "eslint-plugin-tailwindcss";
import eslintHTMLParser from "@html-eslint/parser";
import eslintAngularParser from "@angular-eslint/template-parser";

export default [
{
    "files": [ "**/*.html" ],
    "languageOptions": {
      parser: eslintHTMLParser,
    },
    "plugins": {
      "@html-eslint": eslintHTML,
    },
    "rules": {
     "@html-eslint/require-doctype": 1
    }
},
  {
    "files": [ "**/*.html" ],
    "languageOptions": {
      parser: eslintAngularParser,
    },
    "plugins": {
      "tailwindcss": eslintTailwindCSS,
    },
    "rules": {
        "tailwindcss/no-custom-classname": [ 2 ],
    },
  },
]
```

**Comment by nzakas** [comment]
If you can share a StackBlitz with a repro case, we can take a look.

**Reply by yeonjuan** [reply]
Hi, @nzakas  https://stackblitz.com/~/github.com/yeonjuan/repro-211
This is the process of reproducing.

1. npm install
2. npm run lint
Running eslint will only raise errors corresponding to tailwindcss.
```console
/home/yeonjuan/repro-211/src/index.html
  6:14  error  Classname 'aaa' is not a Tailwind CSS class!  tailwindcss/no-custom-classname
```
3. Comment out the tailwindcss related settings in eslint config.

```js
import eslintHTML from "@html-eslint/eslint-plugin";
import eslintTailwindCSS from "eslint-plugin-tailwindcss";
import eslintHTMLParser from "@html-eslint/parser";
import eslintAngularParser from "@angular-eslint/template-parser";

export default [
{
    "files": [ "**/*.html" ],
    "languageOptions": {
      parser: eslintHTMLParser,
    },
    "plugins": {
      "@html-eslint": eslintHTML,
    },
    "rules": {
     "@html-eslint/require-doctype": 1
    }
  },
  // {
  //   "files": [ "**/*.html" ],
  //   "languageOptions": {
  //     parser: eslintAngularParser,
  //   },
  //   "plugins": {
  //     "tailwindcss": eslintTailwindCSS,
  //   },
  //   "rules": {
  //       "tailwindcss/no-custom-classname": [ 2 ],
  //   },
  // },
]
```

4. npm run lint
Then run eslint and it will output the errors from html-eslint.
```console
/home/yeonjuan/repro-211/src/index.html
  1:1  warning  Missing `<!DOCTYPE HTML>`  @html-eslint/require-doctype
```

**Reply by nzakas** [reply]
The problem is that you're assigning `languageOptions.parser` twice. ESLint combines the two configs so only the last `languageOptions.parser` is used.

**Reply by yeonjuan** [reply]
@nzakas So, is it by design that we can't use different parsers for the same file? Or is there a way to set that up?

**Reply by error-four-o-four** [reply]
I haven't been in such a situation yet but my solution would be to create seperate scripts e.g. `npm run lint:html` and `npm run lint:angular` which use seperate eslint configs with different parsers.

**Reply by nzakas** [reply]
@yeonjuan this is the way ESLint is designed. Each file has a single calculated config with one value for each setting, which includes the parser.

**Reply by yeonjuan** [reply]
@nzakas Ok, I see, maybe the flat-config feature could be updated to apply the individual parser if the parser is explicitly specified? This would be inconvenient when applying ESLint Plugins that use different parsers for the same file. If this makes sense, I'd like to create it as a separate github issue, what do you think?

**Reply by nzakas** [reply]
The entire config system (not just the new one, but the old one too) is designed so that every file has exactly one calculated config. Changing that would require rewriting basically everything to allow for that, and it doesn't make sense to go through that effort for an edge case.

**Reply by xenobytezero** [reply]
I hit exactly this today trying to lint `package.json` with an existing plugin and my own custom plugin with new rules. The existing plugin sets the parser, and my plugin sets the language, which conflict in a way that was difficult to debug. 

I disagree a little that this is an edge case, but if that is the architecture then not much can be done about that. Is there a recommendation for how we would proceed in this case? Simply run ESLint twice with different configurations?
